persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/ClinicalDataSparkRepository.java

```diff
@@ -0,0 +1,277 @@
+package org.cbioportal.persistence.spark;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.apache.spark.sql.*;
+import org.cbioportal.model.*;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.ClinicalDataRepository;
+import org.cbioportal.persistence.PersistenceConstants;
+import org.cbioportal.persistence.spark.util.ParquetConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.apache.spark.sql.functions.*;
+
+
+@Component
+@Qualifier("clinicalDataSparkRepository")
+public class ClinicalDataSparkRepository implements ClinicalDataRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(ClinicalDataSparkRepository.class);
+    private static final String MUTATION_COUNT = "MUTATION_COUNT";
+    private static final String FRACTION_GENOME_ALTERED = "FRACTION_GENOME_ALTERED";
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfSampleInStudy(String studyId, String sampleId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
```

Comment:
Are these unsupported exceptions because we will not be supporting these calls in spark/parquet?  If that is true, in this PR we should have modifications to [ClinicalDataServiceImpl.java](https://github.com/cBioPortal/cbioportal/blob/master/service/src/main/java/org/cbioportal/service/impl/ClinicalDataServiceImpl.java) which can maintain references to multiple repositories (one mybatis and one spark).  I'm not quite sure the best way to handle this.

Reply:
These methods are not called by StudyViewController and so not in scope for this project - as per my understanding. This PR does contain ClniicalDataServiceImpl.java with both mybatis and spark repos, where only the implemented method uses spark repo. Change in ClniicalDataServiceImplTest.java is also included in this PR as well.